src/Cli/AzureCliLogin.ts

```diff
@@ -24,39 +31,25 @@ export class AzureCliLogin {
                 }
             }
         };
-        await this.executeAzCliCommand("--version", true, execOptions);
+
+        await this.executeAzCliCommand('--version', [], true, execOptions);
         core.debug(`az cli version used:\n${output}`);
 
         this.setAzurestackEnvIfNecessary();
 
-        await this.executeAzCliCommand(`cloud set -n "${this.loginConfig.environment}"`, false);
+        await this.executeAzCliCommand('cloud set', ['-n', `${this.loginConfig.environment}`], false);

```

Comment:
Consider move `cloud set` to `args` as well. See https://github.com/Azure/cli/pull/105#discussion_r1226265447